### DISCUSSION TITLE
Boosted frame and RZ geometry

### DISCUSSION THREAD

**Discussion by spadova-a** [BLAST-WarpX/warpx#4412] [Q&A] [answered]
Hi, I have a quick question - is it possible to use boosted frame and RZ geometry at the same time? 

Also, I have a few questions to the example for the LWFA with boosted frame:
1) why are used boundary conditions "periodic" and "pec" and not "PML"?
2) why is there initialized a particle beam? Do you assume that the injection already happened and the simulation is supposed to just accelerated these electrons? It would be really nice to have a brief description of the physics happening in the examples...
3) is it possible to use particles filter in BTD? If not, it is possible to use full diagnostics in the boosted frame and then somehow transform it to the laboratory frame?

**Comment by ax3l** [comment, marked as answer]
Hi there :wave: 

> Hi, I have a quick question - is it possible to use boosted frame and RZ geometry at the same time?

Yes, e.g. https://github.com/ECP-WarpX/WarpX/blob/23.11/Examples/Tests/btd_rz/inputs_rz_z_boosted_BTD

> why are used boundary conditions "periodic" and "pec" and not "PML"?

They are a bit historic and can be updated. The main reason is that PML are a bit more costly to compute (and no particles should return in the moving window sim anyway).

> why is there initialized a particle beam? Do you assume that the injection already happened and the simulation is supposed to just accelerated these electrons? It would be really nice to have a brief description of the physics happening in the examples...

Correct, some examples show what we call "external injection", from either an earlier LPA stage or an external particle accelerator. This is a common case to study staged laser-wakefield accelerators for future collider design.

> is it possible to use particles filter in BTD?

~~No, not yet but it could be implemented similar to https://github.com/ECP-WarpX/WarpX/pull/4371~~
I'll bring it up tomorrow in the developer meeting.

Update: yes, this works! :)

> ~~If not, it is possible to use full diagnostics in the boosted frame and then somehow transform it to the laboratory frame?~~

~~That is not super easy, because on how BTDs work. For the lab-transform, we collect particles along an iso-temporal plane over simulation steps - a single boosted frame dump does not contain enough information (but one would need many subsequent ones).~~

**Comment by RemiLehe** [comment]
@ax3l I think that particle filters do work with BTD (I got confused with this myself: https://github.com/ECP-WarpX/WarpX/issues/4373) ; it's just in a different part of the code.

**Reply by ax3l** [reply]
Thank you, I got confused as well!

@spadova-a yes, this should already work as in regular diagnostics. We will clean up the docs and code on this (thank you @RemiLehe for volunteering!)

**Comment by spadova-a** [comment]
Hi guys,
thanks for the reply, it’s great that the particle filters work also in BTD, this is really important for us.
Cheers.
